pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/PutFormAnswersRoute.java

```diff
@@ -214,6 +215,14 @@ public Object handle(Request request, Response response) {
                     WorkflowResponse workflowResp = workflowService
                             .suggestNextState(handle, operatorGuid, userGuid, studyGuid, fromState)
                             .map(nextState -> {
+                                //special case for SOMATIC_RESULTS ... need to pass same instance in next state and NOT latest
+                                if (instanceDto.getActivityCode().equalsIgnoreCase("SOMATIC_RESULTS")
+                                        && fromState.matches(nextState)) {
```

Comment:
Workflow state is defined to return same state for SOMATIC_RESULTS and hence this will match